common/js/solid.js

```diff
@@ -0,0 +1,440 @@
+        this.tokenize(this.getFeedbackIconClass(feedback))
+      )
+
+    formGroup.classList
+      .add
+      .apply(
+        formGroup.classList,
+        this.tokenize(this.getValidationClass(feedback))
+      )
+  }
+
+  /**
+   * Password Strength Helpers
+   */
+
+  /**
+   * Get OWASP feedback on the given password. Returns false if the password is empty
+   * @param password
+   * @returns {object|false}
+   */
+  PasswordValidator.prototype.getPasswordStrength = function (password) {
+    if (password === '') {
+      return false
+    }
+    return owaspPasswordStrengthTest.test(password)
+  }
+
+  /**
+   * Get the password strength level based on password strength feedback object given by OWASP
+   * @param passwordStrength
+   * @returns {number}
+   */
+  PasswordValidator.prototype.getStrengthLevel = function (passwordStrength) {
+    if (passwordStrength === false) {
+      return 0
+    }
+    if (passwordStrength.requiredTestErrors.length !== 0) {
+      return 1
+    }
+
+    if (passwordStrength.strong === false) {
+      return 2
+    }
+
+    if (passwordStrength.isPassphrase === false || passwordStrength.optionalTestErrors.length !== 0) {
+      return 3
+    }
+
+    return 4
+  }
+
+  PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP = []
+  PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[0] = PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[1] = PasswordValidator.prototype.FEEDBACK_ERROR
+  PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[2] = PasswordValidator.prototype.FEEDBACK_WARNING
+  PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[3] = PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[4] = PasswordValidator.prototype.FEEDBACK_SUCCESS
+
+  /**
+   * @returns {string}
+   */
+  PasswordValidator.prototype.getFeedbackFromLevel = function () {
+    return this.LEVEL_TO_FEEDBACK_MAP[this.currentStrengthLevel]
+  }
+
+  PasswordValidator.prototype.LEVEL_TO_PROGRESS_MAP = []
+  PasswordValidator.prototype.LEVEL_TO_PROGRESS_MAP[0] = PasswordValidator.prototype.STRENGTH_PROGRESS_0
+  PasswordValidator.prototype.LEVEL_TO_PROGRESS_MAP[1] = PasswordValidator.prototype.STRENGTH_PROGRESS_1
+  PasswordValidator.prototype.LEVEL_TO_PROGRESS_MAP[2] = PasswordValidator.prototype.STRENGTH_PROGRESS_2
+  PasswordValidator.prototype.LEVEL_TO_PROGRESS_MAP[3] = PasswordValidator.prototype.STRENGTH_PROGRESS_3
+  PasswordValidator.prototype.LEVEL_TO_PROGRESS_MAP[4] = PasswordValidator.prototype.STRENGTH_PROGRESS_4
+
+  /**
+   * Get the CSS class for the meter based on the current level
+   */
+  PasswordValidator.prototype.getStrengthLevelProgressClass = function () {
+    return this.LEVEL_TO_PROGRESS_MAP[this.currentStrengthLevel]
+  }
+
+  PasswordValidator.prototype.addPasswordError = function (error) {
+    if (Array.isArray(error)) {
+      for (var i = 0, ln = error.length; i < ln; i++) {
+        this.addPasswordError(error[i])
+      }
+      return
+    }
+
+    this.errors.push(error)
+  }
```

Comment:
the above can be replaced by

```
this.errors.push(...(Array.isArray(error) ? error : [error]))
```

or just simply eliminating the call and doing the above inline

Reply:
An whole new world opens to me without the need to support IE11 :D

Reply:
Ah right, I am used to transpiling, so I don't know what browsers need what. However, since no transpiling is in effect here, we might need to be more careful.

Reply:
Fixed